daemon/SecretsImpl/secrets.cpp

```diff
@@ -1223,6 +1261,39 @@ void Daemon::ApiImpl::SecretsRequestQueue::handlePendingRequest(
             }
             break;
         }
+        case GetHealthInfoRequest: {
```

Comment:
The reason why it's here is because the static variables it uses are in this file. But if you are OK with duplicating those, then I can move it there.

Reply:
Don't duplicate, just pass the secretsDirPath as a parameter, I think.

One other thing to check: which thread do we "normally" read the salt data etc within?  Is it always the main thread?  I assume so, but please double check this.

Reply:
Done.